src/Elasticsearch/State/CollectionProvider.php

```diff
@@ -60,13 +59,20 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
         $limit = $body['size'] ??= $this->pagination->getLimit($operation, $context);
         $offset = $body['from'] ??= $this->pagination->getOffset($operation, $context);
 
+        $options = $operation->getStateOptions();
+        if (null === $options) {
+            $options = $this->convertDocumentMetadata($this->documentMetadataFactory->create($resourceClass));
+        } elseif (!$options instanceof OptionsInterface) {
+            throw new \LogicException(sprintf('Operation‘s state options must be null or an instance of %s, but got %s', OptionsInterface::class, \get_class($options)));
```

Comment:
Can we remove the `elseif` ? We can move this exception in the MetadataFactory if we really want to catch this. 

I'd like to suggest:

```
$options = $operation->getStateOptions() ?? new ElasticsearchStateOptions(index: Inflector::tableize($operation->getShortName()));

// TODO: remove in 4.x
if ($operation->getElasticsearch() && !$operation->getStateOptions()) {
  $options = $this->convertDocumentMetadata($this->documentMetadataFactory->create($resourceClass));
}
```

Reply:
I think we should keep the check on instance of  `$options` because at this step of the code, we know only that `$options` is an instance of `ApiPlatform/State/OptionsInterface`, which does not necessarily have `getIndex()`.

Reply:
I understand the concern, alright what about:


```
$options = $operation->getStateOptions() instanceof Options ? $operation->getStateOptions() : new ElasticsearchStateOptions(index: Inflector::tableize($operation->getShortName()));

// TODO: remove in 4.x
if ($operation->getElasticsearch() && !$operation->getStateOptions()) {
  $options = $this->convertDocumentMetadata($this->documentMetadataFactory->create($resourceClass));
}
```